test.js

```diff
@@ -79,3 +79,41 @@ test('leading option - does not call input function after timeout', async t => {
 
 	t.is(count, 1);
 });
+
+// Factory to create a separate class for each test below
+// * Each test replaces methods in the class with a debounced variant,
+//   hence the need to start with fresh class for each test
+const createFixtureClass = () => class {
+	constructor() {
+		this._foo = fixture;
+	}
+
+	foo() {
+		// If `this` is not preserved by `pDebounce()` or `pDebounce.promise()`,
+		// then `this` will be undefined and accessing `this._foo` will throw.
+		return this._foo;
+	}
+
+	getThis() {
+		// If `this` is not preserved by `pDebounce()` or `pDebounce.promise()`,
+		// then `this` will be undefined.
+		return this;
+	}
+};
+
+[
+	['pDebounce()', pDebounce],
+	['pDebounce().promise()', pDebounce.promise]
+].forEach(([name, debounceFn]) => {
```

Comment:
Move the array to a variable and then use a `for-of` loop. Would make it a bit more readable.

Reply:
done